crates/storage-api/src/promise_table/mod.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use super::{protobuf_storage_encode_decode, Result};
+
+use bytestring::ByteString;
+use futures_util::Stream;
+use restate_types::identifiers::{JournalEntryId, PartitionKey, ServiceId};
+use restate_types::journal::EntryResult;
+use std::future::Future;
+use std::ops::RangeInclusive;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum PromiseState {
+    Completed(EntryResult),
+    NotCompleted(Vec<JournalEntryId>),
```

Comment:
Note to myself: Those are the entries of another invocation's journal of the same service instance which are awaiting the promise and want to be notified in case it completes.